### PR TITLE
Documentation enhanced

### PR DESCRIPTION
**Pull request by jlannoy**
More information about custom configuration of the Freemarker template
engine, with small code examples.

### CODE REVIEW

File: ninja-core/src/site/markdown/documentation/html_templating/implicit_functions.md [hunk 1]
```diff
@@ -146,13 +145,4 @@ To get the token, use the following code
...
-Advanced usage of Freemarker
-----------------------------
-
-Freemarker is the templating language we are using for rendering views. 
-It can do a lot of cool stuff, and you should refer to http://freemarker.org/
-to learn more.
-
-Note that the Freemarker <code>Configuration</code> object can be accessed via your application <code>TemplateEngineFreemarker</code> singleton. According to the FreeMarker documentation, the configuration will be thread-safe once all settings have been set via a safe publication technique. Therefore, consider modifying it only within the <code>configure()</code> method of your application <code>Module</code> singleton.
\ No newline at end of file
+If the token is invalid the use will see a 403 Forbidden error page.
```
Comment (raphaelbauer) [inline ninja-core/src/site/markdown/documentation/html_templating/implicit_functions.md]: the "user"

File: ninja-core/src/site/markdown/documentation/html_templating/custom_configuration.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+<#macro bold>
+	<b><#nested /></b>
+</#macro>
+</pre>
+And in your template :
+<pre class="prettyprint">
+<#import "../layout/functions.ftl.html" as f> 
+<@f.bold>Make me bold</@f.bold>
+</pre>
+This last way should help you making reusable macros when using one or another CSS theme.
```
Comment (raphaelbauer) [inline ninja-core/src/site/markdown/documentation/html_templating/custom_configuration.md]: I am curious... what do you mean by that "CSS theme"?

**Comment by raphaelbauer** [comment]
Very well written... Nice enhancement! Please add also one line to changelog.md. There is afaics one simple typo.. once that's fixed we'll merge... :+1:

**Comment by jlannoy** [comment]
Done. I rewrote the sentence about CSS themes. Hope this helps :)

**Comment by jlannoy** [comment]
(PR updated so that it can be merged in next version)

File: ninja-core/src/site/markdown/documentation/html_templating/custom_configuration.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+		...
+	}
+
+}
+</pre>
+
+Custom functions - The Java way
+---------------------------------
+
+While you can access the Freemarker <code>Configuration</code> object, you can add you own custom functions using Java code:
```
Comment (raphaelbauer) [inline ninja-core/src/site/markdown/documentation/html_templating/custom_configuration.md]: add you"r" own...

**Comment by raphaelbauer** [comment]
Very nice (just a tiny spelling mistake)! Will be merged asap...

**Comment by jlannoy** [comment]
(Spelling mistake fixed)

**Comment by raphaelbauer** [comment]
Perfect :) Good work!

**Pull request merged**